src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -141,63 +109,50 @@ void GcsActorManager::RegisterActor(
 void GcsActorManager::ReconstructActorOnWorker(const ray::ClientID &node_id,
                                                const ray::WorkerID &worker_id,
                                                bool need_reschedule) {
-  std::shared_ptr<GcsActor> actor;
-  // Cancel the scheduling of the related actor.
-  auto actor_id = gcs_actor_scheduler_->CancelOnWorker(node_id, worker_id);
-  if (!actor_id.IsNil()) {
-    auto iter = registered_actors_.find(actor_id);
-    RAY_CHECK(iter != registered_actors_.end());
-    actor = iter->second;
-  } else {
-    // Find from worker_to_created_actor_.
-    auto iter = worker_to_created_actor_.find(worker_id);
-    if (iter != worker_to_created_actor_.end()) {
-      actor = std::move(iter->second);
-      // Remove the created actor from worker_to_created_actor_.
-      worker_to_created_actor_.erase(iter);
-      // remove the created actor from node_to_created_actors_.
-      auto node_iter = node_to_created_actors_.find(node_id);
-      RAY_CHECK(node_iter != node_to_created_actors_.end());
-      RAY_CHECK(node_iter->second.erase(actor->GetActorID()) != 0);
-      if (node_iter->second.empty()) {
-        node_to_created_actors_.erase(node_iter);
-      }
+  ActorID actor_id;
+  // Find from worker_to_created_actor_.
+  auto iter = created_actors_.find(node_id);
+  if (iter != created_actors_.end() && iter->second.count(worker_id)) {
+    actor_id = iter->second[worker_id];
```

Comment:
Thanks for the suggestion! Normally I agree but in this case I think it's a bit nicer to put the conditions together so that we only have to call the else block once.